pkg/service/model_build_listener.go

```diff
@@ -234,3 +256,24 @@ func (t *defaultModelBuildTask) buildListenerAttributes(ctx context.Context, svc
 	}
 	return attributes, nil
 }
+
+func mergeServicePortsForListener(ports []corev1.ServicePort) corev1.ServicePort {
```

Comment:
this works but have two issues:
1. we are hacking the k8s `corev1.ServicePort` concept by introduce a new `corev1.Protocol("TCP_UDP")`, which don't exists in k8s, and in theory is an malformed `corev1.ServicePort. (TCP_UDP is not a validate Protocol enum)
2. this don't generate a warning/error if unsupported protocol/protocol conbination is found.
3. this exists as a "glue" code to make the existing `buildListeners` function happy, but in the long run would make the code harder to understand and maintain.


Instead, i think a better option would be:
1. modify buildListeners to accept a list of corev1.ServicePort. inside buildListeners, it first validate whether this list of corev1.ServicePort can form a single listener(and error out when cannot), and generate listener of correct protocol.

Reply:
I'm confused, that's what `buildListeners` now does. It loops through the `corev1.ServicePort`s and determines if there's a case where the same port number exists for `TCP` and `UDP`. It then synthesises a single `corev1.ServicePort` from the original two that uses the `TCP_UDP` protocol that AWS load balancers support.

Either way it then calls `buildListener`.

I could change everything from `buildListener` on down to use an internal data structure that includes a slice of `corev1.ServicePort`s as well as the protocol being used for the load balancer.  That seems like a rather complex change. And it will still have the rats nest of conditionals the current code has. Plus all `corev1.ServicePort` stuff would operate off the first element - so might as well just include a single `corev1.ServicePort`. And at that point all I've done is avoided abusing an enum.

That said, point number two is definitely valid and I'm now returning errors for a few scenarios up the chain.